README.md

```diff
@@ -7,290 +7,505 @@
-- **secret_base64_encoded**: When `true`, it will decode the secret used to verify the token signature. Defaults to `true` (used only for `HS256`).
+# Audience for Management API calls (not required for authentication calls)
+AUTH0_MANAGEMENT_AUDIENCE="https://${AUTH0_DOMAIN}/api/v2/"
+
+# API token for accessing the Management API (not required for authentication calls)
+AUTH0_MANAGEMENT_API_TOKEN="App token goes here"
+```
 
-### Oauth2 authentication
+In your application below the Composer autoload `require`, add:
 
 ```php
-require __DIR__ . '/vendor/autoload.php';
+// ... other use declarations
+use josegonzalez\Dotenv\Loader;
 
-use Auth0\SDK\Auth0;
+// Setup environment vars
+$Dotenv = new Loader(__DIR__ . '/.env');
+$Dotenv->parse()->putenv(true);
+
+// Get environment variables
+echo 'My Auth0 domain is ' . getenv('AUTH0_DOMAIN') . '!';
+```
+
+## Usage - Authentication API
+
+This SDK provides easy-to-implement methods to access the [Authentication API](https://auth0.com/docs/api/authentication). Some common authentication operations are explained below with examples. For additional information and capabilities, please see the methods in the `\Auth0\SDK\API\Authentication` class. Avoid using any methods marked `@deprecated` as they will be removed in the next major version and may not be enabled for your account.
+
+The examples below assume that you followed the installation steps above and are using a `.env` file and loader to store credentials.
+
+### Login
+
+The simplest and most secure way to handle logins to a web application is to use the [Authentication Code grant](https://auth0.com/docs/api-auth/tutorials/authorization-code-grant) combined with Auth0's Universal Login page. In short, that process is:
 
-$domain        = 'YOUR_NAMESPACE';
-$client_id     = 'YOUR_CLIENT_ID';
-$client_secret = 'YOUR_CLIENT_SECRET';
-$redirect_uri  = 'http://YOUR_APP/callback';
+1. An user requesting access is redirected to the Universal Login Page
+2. The user authenticates using one of many possible connections - social like Twitter or Facebook; database using a email and password; passwordless using email or a mobile device
+3. The user is redirected back to your application's callback URL with a `code` and `state` parameter if successful or an `error` and `error_description` if not
+4. If the authentication was successful, the `state` parameter is validated
+5. If the `state` is valid, the `code` parameter is exchanged with Auth0 for an access token
+6. If the exchange is successful, the access token is used to call an Auth0 `/userinfo` endpoint, which returns the now-authenticated user's information
+7. This information can be used to create an account, start an application-specific session, or simply persist as the user session. 
 
+Most of the steps above are handled internally by this SDK. Your application will simply need to:
+
+1. Determine a login action (click a link, visit walled content, etc) and call `Auth0::login()`
```

Comment:
Suggested change: Determine a log in action (for example: click a link, visit walled content, etc.) and call `Auth0::login()`.